proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,624 @@
+
+* Create an entry in the validator substore as:
+  * `storeKey`: `address`, where `address` is the address of the sender of `trs`.
+  * `storeValue`: The serialization of the object `validatorObject` following `validatorObjectSchema` with `validatorObject.name = trs.params.name`. 	
+* Create an entry in the name substore as:
+  * `storeKey`: `trs.params.name` serialized as a utf-8 encoded string. 
+  * `storeValue`: The serialization of the object `validatorAddress` following `validatorAddressSchema` with `validatorAddress.address = address` where `address` is the address of the sender of `trs`.
+* Call `registerValidatorKeys(address, trs.params.proofOfPossession, trs.params.generatorKey, trs.params.blsKey)`, where `address` is the 20-byte address derived from `trs.senderPublicKey`.
+The function `registerValidatorKeys` is defined in the [Validators module][lip-validators-module-register-validator-keys].
+
+#### Update Generator Key Command
+
+This command is used to update the generator key (from [the Validators module][validatorsModule]) for a specific authority. 
+The command ID of this transaction is `COMMAND_ID_UPDATE_KEY`.
+
+##### Parameters
+
+```java
+updateGeneratorKeyParamsSchema = {
+  "type": "object",
+  "properties": {
+    "generatorKey": {
+      "dataType": "bytes",
+      "fieldNumber": 1
+    }
+  },
+  "required": ["generatorKey"]
+}
+```
+
+##### Verification
+
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_UPDATE_KEY` to be executed.
+The list of verification conditions for `trs.params` is as follows:
+
+* Let `address` be the 20-byte address derived from `trs.senderPublicKey`.
+Then the validators substore must have an entry for the store key `address`.
+* `trs.params.generatorKey` must have length 32.
+
+##### Execution
+
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_UPDATE_KEY` to be executed.
+Then `trs.params` implies the following execution logic:
+
+Let `address` be the 20-byte address derived from `trs.senderPublicKey`. 
+Then, call `setValidatorGeneratorKey(address, trs.params.generatorKey)`, where `setValidatorGeneratorKey` is the function exposed by [the Validators module][validatorsModule].
+
+#### Update Authority Command
+
+The command ID for this command is `COMMAND_ID_UPDATE_AUTHORITY`.
+
+##### Parameters
+
+``` java
+updateValidatorParams = {
+   "type": "object",
+   "properties": {
+      "newValidators: {
+          "type": "array",
+          "fieldNumber": 1,    
+          "items": {
+              "type": "object",
+              "required": ["address", "weight"],
+              "properties": {
+                  "address": {
+                      "dataType": "bytes",
+                      "fieldNumber": 1
+                  }
+                  "weight": {
+                      "dataType": "bytes",
+                      "fieldNumber": 2
+                  }
+	      }
+	  }
```

Comment:
The indentation above is also incorrect. Also in this LIP sometimes the indentation is 3 spaces and sometimes 4. Not sure which editor you use, but I would recommend to set the indentation width and type in the editor (4 spaces).

Reply:
There are also some other bugs in the schema:
- missing quotation mark in `"newValidators: `
- missing comma after the address property

It would suggest to copy schema to a file `schema.json` and run
```
 prettier --write --tab-width 4 schema.json 
```